code/controllers/subsystem/mapping.dm

```diff
@@ -0,0 +1,114 @@
+// Subsystem for things such as vaults, away mission init, and procgen.
+
+var/datum/subsystem/mapping/SSmapping
+
+
+/datum/subsystem/mapping
+	name       = "Map"
+	init_order = SS_INIT_MAP
+	flags      = SS_NO_FIRE
+
+	///All possible biomes in assoc list as type || instance
+	var/list/biomes = list()
+
+/datum/subsystem/mapping/New()
+	NEW_SS_GLOBAL(SSmapping)
+
+
+/datum/subsystem/mapping/Initialize(timeofday)
+	if (config.enable_roundstart_away_missions)
+		log_startup_progress("Attempting to generate an away mission...")
+		createRandomZlevel()
+
+	var/watch
+	if (!config.skip_fixedvault_generation)
+		watch = start_watch()
+		log_startup_progress("Placing fixed space structures...")
+		generate_fixedvaults()
+		log_startup_progress("Finished placing fixed structures in [stop_watch(watch)]s.")
+	else
+		log_startup_progress("Not generating fixed vaults - SKIP_VAULT_GENERATION found in config/config.txt")
+
+	if (!config.skip_vault_generation)
+		watch = start_watch()
+		log_startup_progress("Placing random space structures...")
+		generate_vaults()
+		generate_asteroid_secrets()
+		make_mining_asteroid_secrets() // loops 3 times
+		log_startup_progress("Finished placing structures in [stop_watch(watch)]s.")
+	else
+		log_startup_progress("Not generating vaults - SKIP_VAULT_GENERATION found in config/config.txt")
+
+	//hobo shack generation, one shack will spawn, 1/3 chance of two shacks
+	generate_hoboshack()
+	if (rand(1,3) == 3)
```

Comment:
why not `prob(33)` ???

Reply:
This shows as changed simply because I renamed the file. I will sneakycode this fix.